docs/lib/dynamic_config.md

```diff
@@ -0,0 +1,579 @@
+
+On top of Dynaconf, Django Ansible Base also defines a set of standards for file locations, environment variable naming, and settings validation.
+
+
+## Integrating a DAB service with Dynaconf
+
+
+Django settings lifecycle starts by loading the `DJANGO_SETTINGS_MODULE` module, which is usually the `settings` module inside the application, that is the place where the Dynaconf object is created and the settings manipulated to be exported back to Django settings.
+
+That could be done by creating Dynaconf instance directly on the Django app setttings file,
+but DAB helpers are provided to make this process easier and more standardized.
+
+Start by importing the functions from `ansible_base.lib.dynamic_config`:
+
+`myapp/settings.py`:
+```python
+from ansible_base.lib.dynamic_config import (
+    factory,  # Create a dynaconf object
+    export,  # Export dynaconf settings back to django.conf.settings
+    load_envvars, # Load settings from environment variables
+    load_standard_settings_files, # Load settings from standard locations
+    validate, # Validate settings according to a set of validators registered
+)
+```
+
+Those functions are provided separately to allow the application to choose which ones to use and which
+order to call them.
+
+### Creating the Dynaconf object
+
+Assuming `DJANGO_SETTINGS_MODULE` is set to `myapp.settings`:
+
+On the `settings` module (`settings.py` or `settings/__init__.py`), create an object named 
+`DYNACONF` using the `ansible_base.lib.dynamic_config.factory` function.
+
+The `factory` function will accept the following arguments:
+
+- **module_name**: The name of the module
+  The module_name is used to determine the caller module path.
+- **app_name**: The name of the application
+  This name is used to determine some configuration defaults such as the main envvar prefix `{NAME}_` and the standard settings file name on `/etc/ansible-automation-platform/{name}/settings.yaml` and the mode switcher on `{name}_MODE`.
+- **extra_envvar_prefixes**: A list of environment variable prefixes to search for settings
+  If the application needs to load settings filtering additional prefixes, this list can be used to specify them.
+- ****options**: Additional options to pass to dynaconf
+
+
+`myapp/settings.py`:
+```python
+DYNACONF = factory(
+    __name__,
+    "MYAPP",
+    # Options passed directly to dynaconf
+    environments=("development", "production", "testing"),
+    settings_files=["defaults.py"],
+)
+```
+
+With the above configuration DYNACONF will load 
+
+- settings passed directly to the factory function
+- settings from `defaults.py` file
+- settings from `develoment_defaults.py` file if the file exists and the mode is development (which is the default mode)
+- settings from `production_defaults.py` file if the file exists and the mode is production (set with `export MYAPP_MODE=production`)
+- required settings from `django_ansible_base` dynamic configuration
```

Comment:
Is this list ordered? if so, is it correct? i.e. how would a DAB setting get overridden?